lib/plugins/wallet/cardano/api.js

```diff
@@ -0,0 +1,161 @@
+const BN = require('../../../bn')
+
+const unitScale = 6
+
+class CardanoAPI {
+  constructor(client, port, hostname = 'localhost') {
+    this.client = client
+    this.port = port
+    this.hostname = hostname
+  }
+
+  async fetch(endpoint, params, data, appendData = []) {
+    const method = data ? 'POST' : 'GET'
+
+    const url = `https://${this.hostname}:${this.port}/api/v1/${endpoint}`
+
+    return Promise.resolve(true)
+      .then(() => {
+        return this.client({
+          method,
+          url,
+          params,
+          data
+        })
+      })
+      .then(response => {
+        if (response.error) throw r.error
+
+        const responseData = Array.isArray(response.data.data)
+          ? [...response.data.data, ...appendData]
+          : response.data.data
+
+        if (
+          method === 'GET' &&
+          response.data.meta.pagination.totalPages >
+            response.data.meta.pagination.page
+        ) {
+          return this.fetch(
+            endpoint,
+            {
+              ...params,
+              page: (params.page || 1) + 1
+            },
+            data,
+            responseData
+          )
+        }
+
+        return responseData
+      })
+  }
+
+  async walletBalance(walletId, index, confirmations = 1) {
+    return this.fetch('transactions', {
+      walletId,
+      accountIndex: index
+    })
+      .then(transactions =>
+        transactions.reduce((acc, item) => {
```

Comment:
this is a rather complicated reduce. Maybe let's just do `.filter`, then `.map` and finally `.reduce` (or equivalently `_.sum`)

Reply:
ok, that's make sense